stac_geoparquet/arrow/_to_arrow.py

```diff
@@ -1,48 +1,33 @@
 """Convert STAC data into Arrow tables"""
 
-import json
-from datetime import datetime
 from pathlib import Path
 from typing import (
     Any,
     Dict,
-    Generator,
     Iterable,
     Iterator,
-    List,
     Optional,
-    Sequence,
     Union,
 )
 
-import ciso8601
-import numpy as np
 import pyarrow as pa
-import pyarrow.compute as pc
-import shapely
-import shapely.geometry
 
-from stac_geoparquet.arrow._schema.models import InferredSchema
-from stac_geoparquet.arrow._crs import WGS84_CRS_JSON
-from stac_geoparquet.arrow._util import stac_items_to_arrow
-
-
-def _chunks(
-    lst: Sequence[Dict[str, Any]], n: int
-) -> Generator[Sequence[Dict[str, Any]], None, None]:
-    """Yield successive n-sized chunks from lst."""
-    for i in range(0, len(lst), n):
-        yield lst[i : i + n]
+from stac_geoparquet.json_reader import read_json
+from stac_geoparquet.arrow._util import (
+    stac_items_to_arrow,
+    batched_iter,
+    update_batch_schema,
+)
 
 
-def parse_stac_items_to_arrow(
-    items: Sequence[Dict[str, Any]],
+def parse_stac_items_to_batches(
+    items: Iterable[Dict[str, Any]],
     *,
     chunk_size: int = 8192,
-    schema: Optional[Union[pa.Schema, InferredSchema]] = None,
-    downcast: bool = False,
-) -> pa.Table:
-    """Parse a collection of STAC Items to a :class:`pyarrow.Table`.
+    schema: Optional[pa.Schema] = None,
+    downcast: bool = True,
```

Comment:
I'm not 100% sure whether `downcast` should be `True` by default or not. In particular, this has an issue when a schema is passed in because we're converting a float64 to a float32, and we don't modify the user's passed-in schema to be float32 for the bbox columns.

Reply:
Let's just remove the downcast parameter. It doesn't provide a lot of space savings compared to the rest of the file, and it loses precision when round tripping

Reply:
Done in https://github.com/stac-utils/stac-geoparquet/pull/53/commits/541ad1bd3267a434388654c309d6095256f8f647